src/status_im2/subs/shell.cljs

```diff
@@ -17,44 +19,53 @@
                       (first images)))})))
 
 (defn get-card-content
-  [chat communities]
-  (let [last-message (:last-message chat)]
+  [{:keys [chat communities group-chat? primary-name]}]
+  (let [{:keys [content-type content deleted? outgoing deleted-for-me?] :as last-message}
+        (:last-message chat)]
     (merge
      (when last-message
-       (case (:content-type last-message)
-         (constants/content-type-text
-          constants/content-type-emoji)
+       (cond
+         (or deleted-for-me? deleted?)
          {:content-type constants/content-type-text
-          :data         (get last-message :content)}
+          :data         {:text
+                         (if (or deleted-for-me? outgoing)
```

Comment:
I'm curious of why do we have to include this logic here? is it not stored in the db or elsewhere that the message was deleted? 🤔

Reply:
please can you elaborate, what you mean?

Reply:
I would have expected this to be handled by the chat subscription instead, it seems like the preview message of a card should just pull in data and render it and not have any logic of what to render. Perhaps it is okay as is though.